tests/integration-tests/tests/common/osu_common.py

```diff
@@ -103,11 +105,50 @@ def run_osu_benchmarks(
     return output
 
 
-def render_jinja_template(template_file_path, **kwargs):
-    file_loader = FileSystemLoader(str(os_lib.path.dirname(template_file_path)))
-    env = Environment(loader=file_loader)
-    rendered_template = env.get_template(os_lib.path.basename(template_file_path)).render(**kwargs)
-    logging.info("Writing the following to %s\n%s", template_file_path, rendered_template)
-    with open(template_file_path, "w", encoding="utf-8") as f:
-        f.write(rendered_template)
-    return template_file_path
+def run_osu_benchmarks(
+    osu_benchmarks,
+    mpi_variant,
+    partition,
+    remote_command_executor,
+    scheduler_commands,
+    num_of_instances,
+    region,
+    instance,
+    test_datadir,
+    dimensions,
+):
+    for osu_benchmark_group, osu_benchmark_names in osu_benchmarks.items():
+        for osu_benchmark_name in osu_benchmark_names:
+            dimensions_copy = dimensions.copy()
+            logging.info("Running benchmark %s", osu_benchmark_name)
+            output = run_individual_osu_benchmark(
+                mpi_version=mpi_variant,
+                benchmark_group=osu_benchmark_group,
+                benchmark_name=osu_benchmark_name,
+                partition=partition,
+                remote_command_executor=remote_command_executor,
+                scheduler_commands=scheduler_commands,
+                num_of_instances=num_of_instances,
+                slots_per_instance=fetch_instance_slots(region, instance),
+                test_datadir=test_datadir,
+                timeout=40,
+            )
+            logging.info("Preparing benchmarks %s metrics", osu_benchmark_name)
+            metric_data = []
+            for packet_size, latency in re.findall(r"(\d+)\s+(\d+)\.", output):
+                dimensions_copy.update(
+                    {
+                        "OsuBenchmarkGroup": osu_benchmark_group,
+                        "OsuBenchmarkName": osu_benchmark_name,
+                        "PacketSize": packet_size,
+                    }
+                )
+                metric_data.append(
+                    {
+                        "MetricName": "Latency",
+                        "Dimensions": [{"Name": name, "Value": str(value)} for name, value in dimensions_copy.items()],
+                        "Value": int(latency),
+                        "Unit": "Microseconds",
+                    }
+                )
+                yield metric_data
```

Comment:
Doesn't this continually yield longer and longer lists rather than a single list of metric data?

Reply:
Fixed, thank you!
Interestingly, the CloudWatch Metrics looked perfect.